diff --git a/meson.build b/meson.build
index f263b6e944..6099ed2643 100644
--- a/meson.build
+++ b/meson.build
@@ -38,7 +38,7 @@ conf.set_quoted('RELATIVE_SOURCE_PATH', relative_source_path)
 want_ossfuzz = get_option('oss-fuzz')
 want_libfuzzer = get_option('llvm-fuzz')
 want_fuzzbuzz = get_option('fuzzbuzz')
-if want_ossfuzz + want_libfuzzer + want_fuzzbuzz > 1
+if want_ossfuzz and want_libfuzzer and want_fuzzbuzz
         error('only one of oss-fuzz, llvm-fuzz or fuzzbuzz can be specified')
 endif

